docs/concepts/glossary.md

```diff
@@ -0,0 +1,158 @@
+
+A Flink Application Cluster is a dedicated [Flink Cluster](./glossary#flink-cluster) that only
+executes a single [Flink Job](./glossary#flink-job). The lifetime of the
+[Flink Cluster](./glossary#flink-cluster) is bound to the lifetime of the Flink Job. Formerly
+Flink Application Clusters were also known as Flink Clusters in *job mode*. Compare to
+[Flink Session Cluster](./glossary#flink-session-cluster).
+
+#### Flink Cluster
+
+The distributed system consisting of (typically) one Flink Master process and one or more Flink
+Taskmanagers processes.
+
+#### Event
+
+An event is a statement about a change of the state of the domain modelled by the
+application. Events can be input and/or output of a stream or batch processing application.
+Events are special types of [records](./glossary#Record)
+
+#### ExecutionGraph
+
+see [Physical Graph](./glossary#physical-graph)
+
+#### Function
+
+Functions, or user-defined functions (UDFs), are implemented by the user and encapsulate the
+application logic of a Flink program. Most Functions are wrapped by a corresponding
+[Operator](./glossary#operator).
+
+#### Instance
+
+The term *instance* is used to describe a specific instance of a specific type (usually
+[Operator](./glossary#operator) or [Function](./glossary#function)) during runtime. As Apache Flink
+is mostly written in Java, this corresponds to the definition of *Instance* or *Object* in Java.
+In the context of Apache Flink, the term *parallel instance* is also frequently used to emphasize
+that multiple instances of the same [Operator](./glossary#operator) or
+[Function](./glossary#function) type are running in parallel.
+
+#### Flink Job
+
+A Flink Job is the runtime representation of a Flink program. A Flink Job can either be submitted
+to a long running [Flink Session Cluster](./glossary#flink-session-cluster) or it can be started as a
+self-contained [Flink Application Cluster](./glossary#flink-application-cluster).
+
+#### JobGraph
+
+see [Logical Graph](./glossary#logical-graph)
+
+#### JobManager
+
+JobManagers are one of the components which make up the [Flink Master](./glossary#flink-master). A
+JobManager is responsible for supervising the execution of the [Tasks](./glossary#task) of a single
+job.
+
+#### Logical Graph
+
+A logical graph is a directed graph describing the high-level logic of a stream processing program.
+The nodes are [Operators](./glossary#operator) and the edges indicate input/output-relationships or
+data streams or data sets.
+
+#### Managed State
+
+Managed State describes application state which has been registered with the framework. For
+Managed State, Apache Flink will take care about persistence and rescaling among other things.
+
+#### Flink Master
+
+The Flink Master is the master process of a [Flink Cluster](./glossary#flink-cluster). It consists
+of three main components: resource manager, dispatcher and one [JobManager](./glossary#job-manager)
+per running [Flink Job](./glossary#flink-job).
```

Comment:
irrespective of whether to use "JobManager" or "Flink Master", reading about 3 components and then having "on JM per job" sounds a bit strange

Reply:
three "different" components insteads?

Reply:
yes, that might work

Reply:
Done.